src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/repository/OffenderProgramProfileRepository.kt

```diff
@@ -12,4 +13,16 @@ interface OffenderProgramProfileRepository : JpaRepository<OffenderProgramProfil
     courseActivity: CourseActivity,
     offenderBooking: OffenderBooking
   ): OffenderProgramProfile?
+
+  @Query(
+    """
+    from OffenderProgramProfile opp 
+    where opp.courseActivity.courseActivityId = :courseActivityId
+     and opp.offenderBooking.bookingId = :bookingId 
+     and opp.programStatus = 'ALLOC'"""
```

Comment:
Isn't program status a reference code? Should we be modelling as such?

Reply:
I guess so ! Ill change it